common/domain/replication_queue_test.go

```diff
@@ -619,3 +709,27 @@ func TestPurgeAckedMessages(t *testing.T) {
 		})
 	}
 }
+
+func TestReplicationQueueImpl_purgeProcessor(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockQueue := persistence.NewMockQueueManager(ctrl)
+	rq := NewReplicationQueue(mockQueue, "testCluster", nil, nil).(*replicationQueueImpl)
+	atomic.StoreInt32(&rq.status, common.DaemonStatusInitialized)
+
+	done := make(chan bool)
+	mockQueue.EXPECT().GetAckLevels(gomock.Any()).Return(map[string]int64{}, nil).AnyTimes()
+	mockQueue.EXPECT().DeleteMessagesBefore(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+	go func() {
+		rq.Start()
+		close(done)
+	}()
+
+	rq.Stop()
```

Comment:
`rq.Stop()` would be called before `rq.Start()` inside goroutine. 

What exactly are you trying to validate in this function that is not covered by Start/Stop? Expecting `purgeAckedMessages()`  is called once would be nice